pkg/gds/interceptor.go

```diff
@@ -13,6 +16,19 @@ import (
 func (s *Service) serverInterceptor(ctx context.Context, in interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (out interface{}, err error) {
 	// Track how long the method takes to execute.
 	start := time.Now()
+	panicked := true
+
+	// Recover from panics in the handler.
+	// See: https://github.com/grpc-ecosystem/go-grpc-middleware/blob/4705cb37b9857ad51b4c96ff5a2f3c60afe442cf/recovery/interceptors.go#L21-L37
+	defer func() {
+		if r := recover(); r != nil || panicked {
+			log.WithLevel(zerolog.PanicLevel).
+				Err(fmt.Errorf("%v", r)).
+				Str("stack_trace", string(debug.Stack())).
+				Msg("grpc server has recovered from a panic")
+			err = status.Error(codes.Internal, "an unhandled exception occurred")
+		}
+	}()
 
 	// Check if we're in maintenance mode - status method should still return a full response.
 	if s.conf.Maintenance && info.FullMethod != "/trisa.gds.api.v1beta1.TRISADirectory/Status" {
```

Comment:
If I'm reading this right, does this cause the `recover` handler to get called on a status call in maintenance mode?

Reply:
It will, because the `defer` statement is before the maintenance mode check.